### PR TITLE
I have created new CoordinateService

### PR DESCRIPTION
**Pull request by Zingerd**
I have created CertificateService and moved 

1. getAllUndeliveredOrdersWithLiters(), 
2. addCertificate() 
3. getClusteredCoordsAlongWithSpecified()

from UBSManagementService to CoordinateService

### CODE REVIEW

File: service-api/src/main/java/greencity/service/ubs/CoordinateService.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+     * Method to group orders into clusters including summary litres and specified
+     * coordinates.
+     *
+     * @param specified          - list of {@link CoordinatesDto}.
+     * @param litres             - preferred amount of litres.
+     * @param additionalDistance - additional km to radius.
+     * @return List of {@link GroupedOrderDto} lists.
+     * @author Oleh Bilonizhka
+     */
+    List<GroupedOrderDto> getClusteredCoordsAlongWithSpecified(Set<CoordinatesDto> specified,
```
Comment (thesvider) [inline service-api/src/main/java/greencity/service/ubs/CoordinateService.java]: it's good practice to have @ NonNull annotation for public method parameters

File: service/src/main/java/greencity/service/ubs/CoordinateServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+import greencity.repository.OrderRepository;
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static greencity.constant.ErrorMessage.*;
+
+@org.springframework.stereotype.Service
```
Comment (thesvider) [inline service/src/main/java/greencity/service/ubs/CoordinateServiceImpl.java]: @ Service

File: service/src/main/java/greencity/service/ubs/CoordinateServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+     * {@inheritDoc}
+     */
+    @Override
+    public List<GroupedOrderDto> getAllUndeliveredOrdersWithLiters() {
+        Set<Coordinates> allCoords = addressRepository.undeliveredOrdersCoords();
+        List<Order> allOrders = getAllUndeliveredOrders();
+        List<GroupedOrderDto> allOrdersWithLitres = new ArrayList<>();
+        for (Coordinates temp : allCoords) {
+            int currentCoordinatesCapacity =
+                addressRepository.capacity(temp.getLatitude(), temp.getLongitude());
```
Comment (thesvider) [inline service/src/main/java/greencity/service/ubs/CoordinateServiceImpl.java]: repository method name should be self-descriptive.

File: service/src/main/java/greencity/service/ubs/CoordinateServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+public class CoordinateServiceImpl implements CoordinateService {
+    private final AddressRepository addressRepository;
+    private final OrderRepository orderRepository;
+    private final ModelMapper modelMapper;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<GroupedOrderDto> getAllUndeliveredOrdersWithLiters() {
```
Comment (thesvider) [inline service/src/main/java/greencity/service/ubs/CoordinateServiceImpl.java:33]: `    @Override
    public List<GroupedOrderDto> getAllUndeliveredOrdersWithLiters() {
        Set<Coordinates> allCoords = addressRepository.undeliveredOrdersCoords();
        List<Order> allOrders = getAllUndeliveredOrders();
        List<GroupedOrderDto> allOrdersWithLitres = new ArrayList<>();
        for (Coordinates coordinates : allCoords) {
            int currentCoordinatesCapacity = addressRepository.capacity(coordinates.getLatitude(), coordinates.getLongitude());

            List<OrderDto> currentCoordinatesOrders = allOrders.stream()
                    .filter(o -> o.getUbsUser().getAddress().getCoordinates().equals(coordinates))
                    .map(o -> modelMapper.map(o, OrderDto.class))
                    .collect(Collectors.toList());

            allOrdersWithLitres.add(GroupedOrderDto.builder()
                    .amountOfLitres(currentCoordinatesCapacity)
                    .groupOfOrders(currentCoordinatesOrders)
                    .build());
        }
        return allOrdersWithLitres;
    }`

File: service/src/main/java/greencity/service/ubs/CoordinateServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+        Set<Coordinates> allCoords = addressRepository.undeliveredOrdersCoords();
+        List<Order> allOrders = getAllUndeliveredOrders();
+        List<GroupedOrderDto> allOrdersWithLitres = new ArrayList<>();
+        for (Coordinates temp : allCoords) {
+            int currentCoordinatesCapacity =
+                addressRepository.capacity(temp.getLatitude(), temp.getLongitude());
+            List<Order> currentCoordinatesOrders = allOrders.stream().filter(
+                o -> o.getUbsUser().getAddress().getCoordinates().equals(temp)).collect(Collectors.toList());
```
Comment (thesvider) [inline service/src/main/java/greencity/service/ubs/CoordinateServiceImpl.java]: `o.getUbsUser().getAddress().getCoordinates()`

^ potential NPE

File: service/src/main/java/greencity/service/ubs/CoordinateServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+        }
+        return allOrdersWithLitres;
+    }
+
+    /**
+     * Method finds undelivered orders.
+     *
+     * @return List of {@link Order}
+     */
+    private List<Order> getAllUndeliveredOrders() {
```
Comment (thesvider) [inline service/src/main/java/greencity/service/ubs/CoordinateServiceImpl.java:59]: re-implement this. complicated logic with removal from set

File: service/src/main/java/greencity/service/ubs/CoordinateServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,315 @@
...
+                coordinatesInsideRadiusWithoutSpecifiedCoords.add(temp);
+            }
+        }
+        coordinatesInsideRadiusWithoutSpecifiedCoords.removeAll(result);
+
+        coordinatesInsideRadiusWithoutSpecifiedCoords.sort(getComparatorByDistanceFromCenter(centralCoord));
+        int amountOfLitresToFill = litres - specifiedCoordsCapacity;
+        double fill = 0;
+        int allCoordsCapacity = specifiedCoordsCapacity;
+        for (int i = coordinatesInsideRadiusWithoutSpecifiedCoords.size() - 1; i > -1; i--) {
```
Comment (thesvider) [inline service/src/main/java/greencity/service/ubs/CoordinateServiceImpl.java:300]: this is complicated. need to be re-visited. Please check with requirements.

**Review by Natalia1910 — APPROVED** [review]

**Review by dessdee — APPROVED** [review]

**Review by Max877866 — APPROVED** [review]

**Review by Bohdan-Melnyk — APPROVED** [review]

**Pull request merged**